File: Common/Entity/src/main/java/it/ey/dto/IndicatoreDTO.java
package it.ey.dto;

import it.ey.enums.CodTipologiaIndicatoreEnum;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)

public class IndicatoreDTO extends CampiTecniciDTO {
    private Long id;
    private Long idPiao;
    private Long idEntitaFK;
    private CodTipologiaIndicatoreEnum codTipologiaFK;
    private String denominazione;
    private Long idSubDimensioneFK;
    private Long idDimensioneFK;
    private String unitaMisura;
    private String formula;
    private Long peso;
    private String polarita;
    private Long baseLine;
    private Long consuntivo;
    private String fonteDati;
    private TipologiaAndamentoValoreIndicatoreDTO tipAndValAnnoCorrente;
    private TipologiaAndamentoValoreIndicatoreDTO tipAndValAnno1;
    private TipologiaAndamentoValoreIndicatoreDTO tipAndValAnno2;
    private Boolean rilevante;
    private UlterioriInfoDTO addInfo;
}

File: dfp-piao-fe/src/app/shared/models/classes/indicatore-dto.ts
import { CampiTecniciDTO } from './campi-tecnici-dto';
import { PIAODTO } from './piao-dto';
import { TipologiaAndamentoValoreIndicatoreDTO } from './tipologia-andamento-valore-indicatore-dto';
import { UlterioriInfoDTO } from './ulteriori-info-dto';

export class IndicatoreDTO extends CampiTecniciDTO {
  id?: number;
  idPiao?: number;
  idEntitaFK?: number;
  codTipologiaFK?: string;
  denominazione?: string;
  idSubDimensioneFK?: number;
  idDimensioneFK?: number;
  unitaMisura?: string;
  formula?: string;
  peso?: number;
  polarita?: string;
  baseLine?: number;
  consuntivo?: number;
  fonteDati?: string;
  tipAndValAnnoCorrente?: TipologiaAndamentoValoreIndicatoreDTO;
  tipAndValAnno1?: TipologiaAndamentoValoreIndicatoreDTO;
  tipAndValAnno2?: TipologiaAndamentoValoreIndicatoreDTO;
  rilevante?: boolean;
  addInfo?: UlterioriInfoDTO;
}
